maestrowf/interfaces/script/localpooladapter.py

```diff
@@ -0,0 +1,204 @@
+        return self._extension
+
+    def get_header(self, step):
+        """
+        Generate the header present at the top of Flux execution scripts.
+
+        :param step: A StudyStep instance.
+        :returns: A string of the header based on internal batch parameters and
+                  the parameter step.
+        """
+        return f"#!{self._exec}"
+
+    def get_parallelize_command(self, procs, nodes=None, **kwargs):
+        """
+        Generate an empty parallelize command for local execution.
+
+        :param procs: Number of processors to allocate to the parallel call.
+        :param nodes: Number of nodes to allocate to the parallel call
+                      (default = 1).
+        :returns: A string of the parallelize command configured using nodes
+                  and procs.
+        """
+        return ""
+
+    def submit(self, step, path, cwd, job_map=None, env=None):
+        """
+        Submit a script to the Flux scheduler.
+
+        :param step: The StudyStep instance this submission is based on.
+        :param path: Local path to the script to be executed.
+        :param cwd: Path to the current working directory.
+        :param job_map: A dictionary mapping step names to their job
+                        identifiers.
+        :param env: A dict containing a modified environment for execution.
+        :returns: The return status of the submission command and job
+                  identifier.
+        """
+        try:
+            job_id = self._pool.submit(path, cwd)
+            ret_code = 0
+            submit_status = SubmissionCode.OK
+        except Exception as exception:
+            LOGGER.error("Encountered exception: %s", str(exception))
+            job_id = -1
+            ret_code = -1
+            submit_status = SubmissionCode.ERROR
+
+        return SubmissionRecord(submit_status, ret_code, job_id)
+
+    def check_jobs(self, joblist):
+        """
+        For the given job list, query execution status.
+
+        This method uses the scontrol show job <jobid> command and does a
+        regex search for job information.
+
+        :param joblist: A list of job identifiers to be queried.
+        :returns: The return code of the status query, and a dictionary of job
+                  identifiers to their status.
+        """
+        LOGGER.debug("Joblist type -- %s", type(joblist))
+        LOGGER.debug("Joblist contents -- %s", joblist)
+        if not joblist:
+            LOGGER.debug("Empty job list specified.")
+            return JobStatusCode.OK, {}
+        if not isinstance(joblist, list):
+            LOGGER.debug("Specified parameter is not a list.")
+            if isinstance(joblist, int):
+                LOGGER.debug("Integer found.")
+                joblist = [joblist]
+            else:
+                LOGGER.debug("Unknown type. Returning an error.")
+                return JobStatusCode.ERROR, {}
+
+        status = {jid: State.UNKNOWN for jid in joblist}
```

Comment:
Is `jid` here the same as `job_id` used in other places in this class? As I was reading I was confused for a moment and had to think about it, switching it to be `job_id` would preserve clarity even in a generator.

Reply:
Will be done in my next commit.